src/views/language/LanguageTable.tsx

```diff
@@ -14,8 +14,55 @@ import PopulationWarning from '../common/PopulationWarning';
 import { CodeColumn, EndonymColumn, NameColumn } from '../common/table/CommonColumns';
 import ObjectTable from '../common/table/ObjectTable';
 
+import {
+  computeVitalityMetascore,
+  getAllVitalityScores,
+  getVitalityExplanation,
+  VitalityMeterType,
+} from './LanguageVitalityComputation';
+
 const LanguageTable: React.FC = () => {
   const { languagesInSelectedSource } = useDataContext();
+
+  // Helper functions for vitality display
+  function scoreBucket(score: number | null): 0 | 1 | 2 | 3 {
+    if (score === null) return 0;
+    if (score >= 7) return 3; // strong
+    if (score >= 4) return 2; // medium
+    if (score >= 1) return 1; // low
+    return 0; // none/extinct
+  }
+
+  function bucketIcon(bucket: 0 | 1 | 2 | 3) {
```

Comment:
Since you are returning react component, you should use a functional component rather than just a function. Also since you reuse the bucket time you should make it an enum.
```suggestion
const BucketIcon: React.FC<{ bucket: VitalityBucket }> = ({ bucket }) => {
```